azbfs/url_directory.go

```diff
@@ -69,21 +69,26 @@ func (d DirectoryURL) NewDirectoryURL(dirName string) DirectoryURL {
 
 // Create creates a new directory within a File System
 func (d DirectoryURL) Create(ctx context.Context, recreateIfExists bool) (*DirectoryCreateResponse, error) {
+	return d.CreateWithMetadata(ctx, recreateIfExists, nil)
+}
+
+// Create creates a new directory within a File System
+func (d DirectoryURL) CreateWithMetadata(ctx context.Context, recreateIfExists bool, metadata map[string]string) (*DirectoryCreateResponse, error) {
```

Comment:
Is there some naming convention I should be using here or like an options bag approach possibly in case we want to add more options?

Reply:
For example: We may need to expose permissions as well soon

Reply:
In Track 2, we use Options bags containing pointers for optionals, instead of adding "overloads".

Reply:
Created an options bag